Mathlib/Analysis/Distribution/TemperateGrowth.lean

```diff
@@ -115,6 +116,60 @@ lemma HasTemperateGrowth.const (c : F) :
     Function.HasTemperateGrowth (fun _ : E ↦ c) :=
   .of_fderiv (by simpa using .zero) (differentiable_const c) (k := 0) (C := ‖c‖) (fun x ↦ by simp)
 
+/-- Composition of two temperate growth functions is of temperate growth.
+
+Version where the outer function `g` is only of temperate growth on the image of inner function
+`f`. -/
+theorem HasTemperateGrowth.comp' [NormedAddCommGroup D] [NormedSpace ℝ D] {g : E → F} {f : D → E}
+    {t : Set E} (ht : Set.range f ⊆ t) (ht' : UniqueDiffOn ℝ t) (hg₁ : ContDiffOn ℝ ∞ g t)
+    (hg₂ : ∀ N, ∃ k C, ∃ (_hC : 0 ≤ C), ∀ n ≤ N, ∀ x ∈ t,
+    ‖iteratedFDerivWithin ℝ n g t x‖ ≤ C * (1 + ‖x‖) ^ k)
+    (hf : f.HasTemperateGrowth) : (g ∘ f).HasTemperateGrowth := by
```

Comment:
1. I don't think `_hC` is necessary, is it? You should be able to prove it.
2. More importantly, do you think we should have a `HasTemperateGrowthOn` predicate analogous to `ContinuousOn`? I suspect so. This theorem is analogous to [ContinuousOn.comp_continuous](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/ContinuousOn.html#ContinuousOn.comp_continuous).

Reply:
> About `_hC`, I have just told Moritz that assuming it would be fine, because the proof was a bit painful... I have no strong opinion though, so if you think it's worth it I'm fine with rolling back.

Ah, sorry, I missed that. I'm fine with leaving it as Anatole suggested, although I'll note that it just shifts where you need to prove nonnegativity. The main reason I was advocating for it was for the parallelism with `HasTemperateGrowth` which would make a switch to `HasTemperateGrowthOn` smoother.